databricks_cli/permissions/cli.py

```diff
@@ -0,0 +1,178 @@
+# Databricks CLI
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+
+import click
+from click import UsageError
+
+from databricks_cli.click_types import OneOfOption
+from databricks_cli.configure.config import provide_api_client, profile_option, debug_option
+from databricks_cli.permissions.api import PermissionsApi, PermissionTargets, PermissionLevel, \
+    PermissionType, Permission, PermissionsObject
+from databricks_cli.utils import eat_exceptions, CONTEXT_SETTINGS
+from databricks_cli.utils import pretty_format
+from databricks_cli.version import print_version_callback, version
+from databricks_cli.workspace.api import WorkspaceApi
+
+FILTERS_HELP = 'Filters for filtering the list of users: ' + \
+               'https://docs.databricks.com/api/latest/scim.html#filters'
+USER_OPTIONS = ['user-id', 'user-name']
+JSON_FILE_OPTIONS = ['json-file', 'json']
+CREATE_USER_OPTIONS = JSON_FILE_OPTIONS + ['user-name']
+
+GROUP_USER_SERVICE_OPTIONS = ['group-name', 'user-name', 'service-name']
+PERMISSION_LEVEL_OPTIONS = PermissionLevel.names()
+POSSIBLE_OBJECT_TYPES = 'Possible object types are: \n\t{}\n'.format(
+    PermissionTargets.help_values())
+
+POSSIBLE_PERMISSION_LEVELS = 'Possible permission levels are: \n\t{}\n'.format(
+    PermissionLevel.help_values())
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Get permissions for an item.  ' + POSSIBLE_OBJECT_TYPES)
+@click.option('--object-type', required=True, help=POSSIBLE_OBJECT_TYPES)
+@click.option('--object-id', required=True, help='object id to require permission about')
```

Comment:
why not providing `one of` kind of arguments? e.g. `--cluster-id`, `--job-id`, `--tokens`, `--passwords`, `--cluster-policy-id` and etc? CLI is supposed to simplify usage for admins and not always be 1-1 mapping to API resources.